std/math.d

```diff
@@ -5901,22 +5901,30 @@ public:
             return true;
         else version(AArch64)
         {
+            bool result;
             auto oldState = getControlState();
-            // If exceptions are not supported, we set the bit but read it back as zero
-            // https://sourceware.org/git/?p=glibc.git;a=blob;f=sysdeps/aarch64/fpu/feenablxcpth.c
-            setControlState(oldState | (divByZeroException & allExceptions));
-            immutable result = (getControlState() & allExceptions) != 0;
-            setControlState(oldState);
```

Comment:
`setControlState()` cannot be pure in general, so I extracted the `__asm` instructions from there and, [as long as we're only setting exception bits, `hasExceptionTraps()` can still be considered pure (see point 8)](https://dlang.org/spec/function.html#pure-functions).

Reply:
Actually, as per the concession in the language spec, `setControlState()` _can_ be pure also.  As the FPControl itself saves state in `initialize()` and clears and restores state in `~this()`.

Reply:
Actually, as per the concession in the language spec, `setControlState()` can be pure also. As the FPControl itself saves state in `initialize()` and clears and restores state in `~this()`.

Reply:
These are all static functions though, so the struct is not initialized or deconstructed when they're called?

Reply:
Only `hasExceptionTraps()` is static, all others require a state object.

Reply:
Well, `hasExceptionTraps()` is the function in question, so it cannot rely on the struct resetting these flags.